code/modules/mob/living/recoil.dm

```diff
@@ -43,23 +42,66 @@
 	recoil_timer = null
 	update_recoil_cursor(G)
 
-/mob/living/proc/update_recoil_cursor(var/obj/item/weapon/gun/G)
-	G.update_cursor(src)
-	var/bottom = 0
-	switch(recoil)
-		if(0 to 10)
-			;
-		if(10 to 20)
-			bottom = 10
-		if(20 to 30)
-			bottom = 20
-		if(30 to 50)
-			bottom = 30
-		if(50 to MAX_ACCURACY_OFFSET)
-			bottom = 50
-		if(MAX_ACCURACY_OFFSET to INFINITY)
-			bottom = MAX_ACCURACY_OFFSET
-	if(bottom)
-		var/reduction = calc_reduction()
-		if(reduction > 0)
-			recoil_timer = addtimer(CALLBACK(src, .proc/update_recoil_cursor, G), 1 + (recoil - bottom) / reduction)
+/mob/living/proc/update_recoil_cursor()
+	update_cursor()
+	var/timer = calc_timer()
+	if(timer > 0)
+		deltimer(recoil_timer)
+		recoil_timer = addtimer(CALLBACK(src, .proc/update_recoil_cursor), timer, TIMER_STOPPABLE)
+
+/mob/living/proc/update_cursor()
+	if(get_preference_value(/datum/client_preference/gun_cursor) != GLOB.PREF_YES)
+		remove_cursor()
+		return
+	if(client)
+		client.mouse_pointer_icon = initial(client.mouse_pointer_icon)
+		var/offset = round(calc_recoil())
```

Comment:
```suggestion
		var/offset = min(round(calc_recoil()), MAX_ACCURACY_OFFSET)
```
Apparently recoil can be higher than `MAX_ACCURACY_OFFSET` and it causes next assert to fail.

Reply:
How were you able to do this?

Reply:
By shooting a full mag from Wintermute on full auto. Not that hard.

Reply:
What is the object code for that?

Reply:
I am not sure what code you are asking me for. Anyway, I have looked into the problem and found out that recoil should be capped by `MAX_ACCURACY_OFFSET`, but the capping code https://github.com/discordia-space/CEV-Eris/blob/57b2c11fe180717b7156be213c8cea197381225b/code/modules/projectiles/gun.dm#L349 was lost in #4313. It should be added back to `/mob/living/proc/handle_recoil`.

Reply:
The object path, sorry. Like /obj/item/weapon/projectile/lmg or something like that

Reply:
That is `/obj/item/weapon/gun/projectile/automatic/IH_heavyrifle`.